### DISCUSSION TITLE
Having messages in seperate JSON files.

### DISCUSSION THREAD

**Discussion by codeflorist** [intlify/vue-i18n#1232] [General]
All docs of this library and the nuxt module show the defining of the translation messages either inside the `VueI18n.createI18n` call (put inside a `main.js`) or the nuxt config file `nuxt.config.js`.

This is in my humble opinion **IN NO WAY** recommendable. Translation messages need to be put inside seperate JSON files inside a `locales` directory. About every other translation library does it that way. It's the industry standard to do it this way.

Why are the docs of `vue-i18n` and `@nuxtjs/i18n` propagating everywhere to put translation messages in `main.js` files - or even worse - inside config files?!

Heck - i cannot even find ANY basic information on the docs pages on how to use seperate JSON-files. Where can i find that info?

**Comment by Damian-Orzel** [comment]
Any update?

**Reply by codeflorist** [reply]
check out this documentation: https://vue-i18n.intlify.dev/guide/advanced/lazy.html
